### ISSUE TITLE
Retrieval of failed message body can take a long time

### ISSUE THREAD

**Issue by johnsimons** [Particular/ServicePulse#307]
In the following screen, the retrieval of large bodies can take a long time, and make the UI unresponsive.
![image](https://cloud.githubusercontent.com/assets/122651/13972727/2384a502-f0e7-11e5-9524-3124b349a069.png)

My proposal is to check the message body size and then if the message size is over a threshold, we don't show it inline and instead present the user with a url to d/l it directly.

**Comment by johnsimons** [comment]
@petersgiles as discusssed

**Comment by sergioc** [comment]
It would always make more sense to [show message data in a modal](https://github.com/Particular/ServicePulse/issues/308), where we're able to add loading indicators and control the feedback to the user. This will make rendering of the message body consistent, regardless of message size.

@johnsimons would that be an appropriate solution for this specific scenario?

**Comment by johnsimons** [comment]
I don't think so @sergioc, IMO any body greater then 256K is too big to show on the browser, it actually makes the browser slow.

**Comment by sergioc** [comment]
Why does it make the browser slow?

**Comment by WojcikMike** [comment]
On some devices displaying 1 MB file in a browser can slow down the browser as this is a lot of text. As our current design show body on a list itself I would suggest showing only first 256 chars (or any other round number) and allow to dowload the whole body as a text file. 

In this case if the body is big I would most likely open it in some text editor for better search and formatting.
